(master) App crashes while setting background

**Describe the bug**
Sometimes the app will crash while it's loading background images. I can't really find a pattern to it, but I've noticed it more when I scroll around fast.

**Logs**
```
2021-03-03 19:33:48.116 15414-15502/org.jellyfin.androidtv.debug E/ACRA: ACRA caught a ExecutionException for org.jellyfin.androidtv.debug
    java.util.concurrent.ExecutionException: com.bumptech.glide.load.engine.GlideException: Failed to load resource
    There was 1 root cause:
    com.bumptech.glide.load.HttpException(Failed to connect or obtain data, status code: -1)
     call GlideException#logRootCauses(String) for more detail
        at com.bumptech.glide.request.RequestFutureTarget.doGet(RequestFutureTarget.java:217)
        at com.bumptech.glide.request.RequestFutureTarget.get(RequestFutureTarget.java:130)
        at org.jellyfin.androidtv.data.service.BackgroundService$setBackground$1$backdropDrawables$2$1.invokeSuspend(BackgroundService.kt:161)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
     Caused by: com.bumptech.glide.load.engine.GlideException: Failed to load resource
    There was 1 root cause:
    com.bumptech.glide.load.HttpException(Failed to connect or obtain data, status code: -1)
     call GlideException#logRootCauses(String) for more detail
```

**System (please complete the following information):**
 - Emulator, Fire TV Stick, Sony Android TVs

Got it again while connecting directly to the IP address over http:
```
2021-03-03 20:59:23.820 20543-20622/org.jellyfin.androidtv.debug E/ACRA: ACRA caught a ExecutionException for org.jellyfin.androidtv.debug
    java.util.concurrent.ExecutionException: com.bumptech.glide.load.engine.GlideException: Failed to load resource
    There was 1 root cause:
    com.bumptech.glide.load.HttpException(Failed to connect or obtain data, status code: -1)
     call GlideException#logRootCauses(String) for more detail
        at com.bumptech.glide.request.RequestFutureTarget.doGet(RequestFutureTarget.java:217)
        at com.bumptech.glide.request.RequestFutureTarget.get(RequestFutureTarget.java:130)
        at org.jellyfin.androidtv.data.service.BackgroundService$setBackground$1$backdropDrawables$2$1.invokeSuspend(BackgroundService.kt:161)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
     Caused by: com.bumptech.glide.load.engine.GlideException: Failed to load resource
    There was 1 root cause:
    com.bumptech.glide.load.HttpException(Failed to connect or obtain data, status code: -1)
     call GlideException#logRootCauses(String) for more detail
```

--- a/app/src/main/java/org/jellyfin/androidtv/data/service/BackgroundService.kt
+++ b/app/src/main/java/org/jellyfin/androidtv/data/service/BackgroundService.kt
@@ -19,6 +19,7 @@ import androidx.core.graphics.drawable.toDrawable
 import androidx.fragment.app.FragmentActivity
 import androidx.window.WindowManager
 import com.bumptech.glide.Glide
+import com.bumptech.glide.load.engine.GlideException
 import kotlinx.coroutines.*
 import org.jellyfin.androidtv.R
 import org.jellyfin.androidtv.preference.UserPreferences
@@ -28,7 +29,6 @@ import org.jellyfin.apiclient.model.dto.ImageOptions
 import org.jellyfin.apiclient.model.entities.ImageType
 import org.jellyfin.apiclient.model.search.SearchHint
 import timber.log.Timber
-import java.lang.Exception
 
 class BackgroundService(
 	private val context: Context,
@@ -185,13 +185,14 @@ class BackgroundService(
 				.map { future -> async {
 					try {
 						future.get()
-					} catch (ex: Exception) {
+					} catch (ex: GlideException) {
+						Timber.e(ex, "There was an error fetching the background image:")
 						null
 					}
 				} }
 				.awaitAll()
-				.onEach { it?.colorFilter = colorFilter }
 				.filterNotNull()
+				.onEach { it.colorFilter = colorFilter }
 
 			backgrounds.clear()
 			backgrounds.addAll(backdropDrawables)

--- a/app/src/main/java/org/jellyfin/androidtv/data/service/BackgroundService.kt
+++ b/app/src/main/java/org/jellyfin/androidtv/data/service/BackgroundService.kt
@@ -182,14 +182,16 @@ class BackgroundService(
 						.centerCrop()
 						.submit()
 				}
-				.map { future -> async {
-					try {
-						future.get()
-					} catch (ex: GlideException) {
-						Timber.e(ex, "There was an error fetching the background image:")
-						null
+				.map { future ->
+					async {
+						try {
+							future.get()
+						} catch (ex: GlideException) {
+							Timber.e(ex, "There was an error fetching the background image.")
+							null
+						}
 					}
-				} }
+				}
 				.awaitAll()
 				.filterNotNull()
 				.onEach { it.colorFilter = colorFilter }

--- a/app/src/main/java/org/jellyfin/androidtv/data/service/BackgroundService.kt
+++ b/app/src/main/java/org/jellyfin/androidtv/data/service/BackgroundService.kt
@@ -28,6 +28,7 @@ import org.jellyfin.apiclient.model.dto.ImageOptions
 import org.jellyfin.apiclient.model.entities.ImageType
 import org.jellyfin.apiclient.model.search.SearchHint
 import timber.log.Timber
+import java.lang.Exception
 
 class BackgroundService(
 	private val context: Context,
@@ -53,7 +54,7 @@ class BackgroundService(
 	// Current background index
 	private var currentIndex = 0
 
-	// Prefered display size, set when calling [attach].
+	// Preferred display size, set when calling [attach].
 	private var windowSize = Size(0, 0)
 	private var windowBackground: Drawable = ColorDrawable(Color.BLACK)
 
@@ -181,9 +182,15 @@ class BackgroundService(
 						.centerCrop()
 						.submit()
 				}
-				.map { future -> async { future.get() } }
+				.map { future -> async {
+					try {
+						future.get()
+					} catch (ex: Exception) {
+						null
+					}
+				} }
 				.awaitAll()
-				.onEach { it.colorFilter = colorFilter }
+				.onEach { it?.colorFilter = colorFilter }
 				.filterNotNull()
 
 			backgrounds.clear()

